map.js

```diff
@@ -467,66 +415,108 @@ function initListeners(){
             moveMap({x:event.movementX, y:event.movementY});
         }
         // regardless of whether we are down or not, we need to redraw the scene?
-        // TODO: only redraw if we move on to or off of an icon.
+        // TODO: only redraw if we dragged or move on to or off of an icon?
         drawFrame();
     });
     viewport.addEventListener("mouseup", function(event){
         lastMouseLoc = new Point(event.offsetX, event.offsetY);
         isDown = false;
         //yay we get to interpret clicks on our own! /s
```

Comment:
The default onClick() basically happens after every mouseUp(). So you can mousedown, then drag for 5 seconds, then mouseup, and it'll call onClick(). we don't want that, so we have to create our own onClick detecction + handler